aiida/orm/nodes/node.py

```diff
@@ -885,22 +889,37 @@ def _iter_all_same_nodes(self, allow_before_store=False) -> Iterator['Node']:
         """
         if not allow_before_store and not self.is_stored:
             raise exceptions.InvalidOperation('You can get the hash only after having stored the node')
+
         node_hash = self._get_hash()
 
         if not node_hash or not self._cachable:
             return iter(())
 
         builder = QueryBuilder(backend=self.backend)
-        builder.append(self.__class__, filters={'extras._aiida_hash': node_hash}, project='*', subclassing=False)
-        nodes_identical = (n[0] for n in builder.iterall())
+        builder.append(self.__class__, filters={f'extras.{self._HASH_EXTRA_KEY}': node_hash}, subclassing=False)
+
+        return (node for node in builder.all(flat=True) if node.is_valid_cache)
+
+    def set_is_valid_cache(self, valid: bool) -> None:
```

Comment:
I like that, would make it easier for `Node` subclasses to override the behavior. Maybe we should document that as an "official" way to influence caching?

Reply:
It certainly is possible and probably the right thing to do. Only reason I already didn't do it was because originally the method was called something completely different. Then I saw that `is_valid_cache` exists and that is essentially the getter of what I was implementing, however, that was currently being documented as a property to override by _classes_ (not _instances_). So then it felt a bit weird to make the setter really only operate on instances.

But then again, that is what the property is now doing: it is now an instance level concept, and so your suggestion makes sense. Will change it.

Reply:
> Maybe we should document that as an "official" way to influence caching?

If people agree that this is indeed the way to go to officially control caching on a per-node basis, then I would definitely add this to the docs. I first wanted to get a sense if people agree before writing this though. So I take @greschd is on board. Then I will just wait a bit longer for the others to have the chance to respond.

Reply:
Done